File: dotnet/windntrees.net/DataAccess/DBFModels/Metadata/Color.cs
using System.ComponentModel.DataAnnotations;
using SharedLibrary.Attributes;

namespace DataAccess.DBFModels.Metadata
{
    [MetadataType(typeof(ColorMD))]
    public partial class Color
    {

    }

    public partial class ColorMD
    {
        [Key]
        public System.Guid UID { get; set; }

        [LocaleMessageRequired]
        [LocaleMessageStringLength(10)]
        [Display(ResourceType = typeof(LocaleResources.Contents.Color), Name = "Name")]
        public string Name { get; set; }

        [LocaleMessageStringLength(255)]
        [Display(ResourceType = typeof(LocaleResources.Contents.Color), Name = "Description")]
        public string Description { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }

    }
}

File: dotnet/windntrees.net/Application/Scripts/ViewModels/Color.js
function Color(data) {
    var instance = this;

    instance._datakey = data.UID;

    instance.UID = data.UID;

    instance.Name = ko.observable(data.Name);
    instance.Name.extend({
        required: true,
        maxLength: 10
    });

    instance.Description = ko.observable(data.Description);
    instance.Description.extend({
        maxLength: 255
    });

    instance.RowVersion = data.RowVersion;
}